samples/ExtendsTypeVariableImplementedForNullableTypeArgument.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 The JSpecify Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+import org.jspecify.annotations.NullnessUnspecified;
+
+@NullMarked
+class ExtendsTypeVariableImplementedForNullableTypeArgument {
+  interface NullableBounded<T extends @Nullable Object> {
+    <U extends T> void x(U u);
```

Comment:
The fact that this signature doesn't really make sense is not an absolute obstacle but is there a version of this test that looks more like something reasonable someone might want to do?

Reply:
Being minimal has its advantages, like how this test checks that we detect the problem even when the mismatch arises only in the parameters. The most reasonable alternative I can think of offhand involves having both a parameter and a return type.

That said, this PR description already theorizes a bunch about how we could minimize this test further and what the actual results would be, and I'm happy not to go down that road. And as you say, real-world examples are nice when you can get them. So here's one, I think.